02-spatial-data.Rmd

```diff
@@ -710,66 +712,62 @@ Similar to the **sf** package, **raster** also provides `plot()` methods for its
-To create an object of class `RasterLayer`, we can use the `raster()` function.
-This function behavior changes based on a given arguments.
-
-For example, a new `RasterLayer` object could be created from a file, when a path to a raster file is provided:
+The easiest way to create a raster object in R is to read-in a raster file from disk or from a server.
 
 ```{r}
 raster_filepath = system.file("raster/srtm.tif", package = "spDataLarge")
 new_raster = raster(raster_filepath)
 ```
 
-Without any arguments, the `raster()` function would create an empty `RasterLayer`:
-
-```{r}
-new_raster2 = raster()
-new_raster2
-```
-
-An existing raster could be used a as a template for a new object when a main argument `x` is of the `Raster*` class (note that `new_raster3` do not keep the values of `new_raster`):
+The **raster** package support numerous drivers with the help of **rgdal**.
+To find out which drivers are available on your system, run `raster::writeFormats()` and `rgdal::gdalDrivers()`.
 
-```{r}
-new_raster3 = raster(new_raster)
-new_raster3
-```
-
-Creation of the `RasterLayer` object is also possible with parameters given by the user.
-In this kind of situations, a number of columns and rows and extent needs to be specified:
+Aside from reading in a raster, you can also create one from scratch.
+Running `raster()` creates an empty `RasterLayer`.
+Here, however, we will create manually a very simple raster.
+This should make it easy to understand how raster and related operations work.
+Our raster should consist of just three rows and columns centered around the null meridian and the equator (see xmn, xmx, ymn and ymx parameters).
+Additionally, we define a resolution of 1, which here corresponds to one degree since the default proj4string of a raster object is WGS84.
+Finally, we set the values with the vals argument.
+Here, we just number the cells, that means we assign 1 to cell 1, 2 to cell 2, and finally 36 to cell 36.
+We know that there are 36 cells by multiplying six (rows) by six (columns).
+As we have seen above, setting raster values corresponds to a rowwise cell filling starting at the upper left corner.
+Consequently, the upper first row contains the values 1 to 6, the second row 7 to 12, and the last row 31 to 36.
 
 ```{r}
 # creation of the RasterLayer object with a given number of columns and rows, and extent
-new_raster4 = raster(ncol = 101, nrow = 77, xmn = 0, xmx = 101, ymn = 0, ymx = 77)
-new_raster4
+new_raster2 = raster(nrow = 6, ncol = 6, res = 0.5, 
+                     xmn = -1.5, xmx = 1.5, ymn = -1.5, ymx = 1.5,
+                     vals = 1:36)
 ```
 
-Finally, this function could create `RasterLayer` based on many types of objects, such as a `matrix`, `Extent` or `Spatial*` (from **sp**) object.
-For more information take a look at the help file - `?raster`.
-
+For still further ways of creating a raster object have a look at the help file - `?raster`.
+<!--
```

Comment:
I have created a raster manually which I will also use in chapter 3. Moreover, I have commented the values, getValues part since I describe these operations in the subsetting and manipulation raster sections. I hope you are ok with it?

Reply:
I agree with your attempt to minimise duplication. Good work. Will leave it to @Nowosad, who wrote this, to 'sign off' on this change. Fine by me though.

Reply:
Fine be me for now as well. We could always decide later where it fits better. 
The only question I have for now is about the new raster dataset. Shouldn't we use something we have now? Or (alternatively) add this new data to our data packages?

Reply:
@Nowosad I created the raster dataset manually, and also described how I did so in the text. I simply replaced the part where you have created a raster object manually with my smaller raster.